tests/print.rs

```diff
@@ -0,0 +1,65 @@
+#[cfg(test)]
+mod tests {
+    use yaiwr::{instruction::Instruction, Calc};
+    #[test]
+    fn eval_println_statement_add() {
+        let ast = Calc::from_str("println(2+2)").unwrap();
+        let bytecode = &mut vec![];
+        Calc::to_bytecode(ast, bytecode);
+        assert_eq!(Calc::eval(bytecode), Ok(None));
+    }
+    #[test]
+    fn eval_println_statement_mul() {
+        let ast = Calc::from_str("println(2*2)").unwrap();
+        let bytecode = &mut vec![];
+        Calc::to_bytecode(ast, bytecode);
+        assert_eq!(Calc::eval(bytecode), Ok(None));
+    }
+
+    #[test]
+    fn println_statement_numeric_bytecode() {
+        let ast = Calc::from_str("println(1)").unwrap();
+        let bytecode = &mut vec![];
+        Calc::to_bytecode(ast, bytecode);
+        match bytecode.as_slice() {
+            [first, second] => {
+                assert_eq!(first, &Instruction::Push { value: 1 });
+                assert_eq!(second, &Instruction::PrintLn);
+            }
+            _ => panic!("expected bytecodes to be not empty!"),
+        }
+    }
+
+    #[test]
+    fn print_statement_add_bytecode() {
+        let ast = Calc::from_str("println (1+1)").unwrap();
+        let bytecode = &mut vec![];
+        Calc::to_bytecode(ast, bytecode);
+        match bytecode.as_slice() {
+            [c1, c2, c3, c4] => {
+                assert_eq!(c1, &Instruction::Push { value: 1 });
+                assert_eq!(c2, &Instruction::Push { value: 1 });
+                assert_eq!(c3, &Instruction::Add {});
+                assert_eq!(c4, &Instruction::PrintLn {});
+            }
+            _ => panic!("expected bytecodes to be not empty!"),
+        }
+    }
+
+    #[test]
+    #[should_panic]
+    fn eval_println_statement_add_parsing_error() {
+        let ast = Calc::from_str("println 2+2").unwrap();
+        let bytecode = &mut vec![];
+        Calc::to_bytecode(ast, bytecode);
+        assert_eq!(Calc::eval(bytecode), Ok(None));
+    }
+    #[test]
+    #[should_panic]
+    fn eval_println_statement_mul_parsing_error() {
+        let ast = Calc::from_str("println 2*2").unwrap();
+        let bytecode = &mut vec![];
+        Calc::to_bytecode(ast, bytecode);
+        assert_eq!(Calc::eval(bytecode), Ok(None));
+    }
```

Comment:
How about a test that spawns a fresh interpreter process and captures the output to check `println` works?

Reply:
I guess there should be a way to execute it as a shell command?

Reply:
Not sure that's the best way to test it cause now my tests are dependent on `cargo` being available on the machine but then maybe it's okay since I will always have cargo along my rust tests :thinking: 
added :point_up: 7fd3122716f95997c0bfdcef8dbf3cf63f617821

Reply:
You can assume cargo is always present. Rust isn't really usable without it, and typically people don't ship the test binaries anyway.